aiida_common_workflows/workflows/relax/bigdft/generator.py

```diff
@@ -0,0 +1,117 @@
+
+__all__ = ('BigDFTRelaxInputsGenerator',)
+
+BigDFTRelaxWorkChain = WorkflowFactory('bigdft.relax')
+BigDFTParameters = DataFactory('bigdft')
+
+
+class BigDFTRelaxInputsGenerator(RelaxInputsGenerator):
+    """Input generator for the `BigDFTRelaxWorkChain`."""
+
+    _default_protocol = 'moderate'
+    _protocols = {
+        'fast': {
+            'description': 'This profile should be chosen if speed\
+ is more important than accuracy.',
+            'inputdict_cubic': {
+                'dft': {
+                    'hgrids': 0.45
+                }
+            },
+            'inputdic_linear': {
+                'import': 'linear_fast'
+            }
+        },
+        'moderate': {
+            'description': 'This profile should be chosen if accurate forces \
+are required, but there is no need for extremely \
+accurate energies.',
+            'inputdict_cubic': {},
+            'inputdict_linear': {
+                'import': 'linear'
+            }
+        },
+        'precise': {
+            'description': 'This profile should be chosen if highly accurate\
+                            energy differences are required.',
+            'inputdict_cubic': {
+                'dft': {
+                    'hgrids': 0.15
+                }
+            },
+            'inputdict_linear': {
+                'import': 'linear_accurate'
+            }
+        }
+    }
+
+    _calc_types = {'relax': {'code_plugin': 'bigdft.bigdft', 'description': 'The code to perform the relaxation.'}}
+
+    _relax_types = {
+        RelaxType.ATOMS: 'Relax only the atomic positions while keeping the cell fixed.',
+    }
+
+    def get_builder(
+        self,
+        structure,
+        calc_engines,
+        protocol,
+        relaxation_type,
+        threshold_forces=None,
+        threshold_stress=None,
+        **kwargs
+    ):
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+
+        :param structure: the structure to be relaxed
+        :param calc_engines: ...
+        :param protocol: the protocol to use when determining the workchain inputs
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        # pylint: disable=too-many-locals
+        from aiida.orm import Dict
+        from aiida.orm import load_code
+
+        #Pseudo fam
+        #pseudo_fam = kwargs.pop('pseudo_family')
+        if relaxation_type == RelaxType.ATOMS:
+            relaxation_schema = 'relax'
+        else:
+            raise ValueError('relaxation type `{}` is not supported'.format(relaxation_type.value))
```

Comment:
This should probably be automated by the base class. Since the supported `relax_types` are defined by the `_relax_types` class attribute, the base class can do this. I will make a PR for this.